### ISSUE TITLE
v_boundaries unhandled DomainError

### ISSUE THREAD

**Issue by michael-petersen** [JuliaStellarDynamics/OrbitalElements.jl#14]
tl;dr: a `DomainError` problem when trying to find αminαmax in `MestelPotential` or `TaperedMestel`

MWE:
```
using OrbitalElements

# set up the resonance
n1,n2 = -1,2

# set up the model
model = TaperedMestel()
rc,rmin,rmax = 1.0,0.2,20. 

O = OrbitalElements.OrbitalParameters(rmin=rmin,rmax=rmax)
resonance = Resonance(n1,n2,model,O)
OrbitalElements.v_boundaries(0.0,resonance,model,O)
# this fails with ERROR: DomainError with -0.0002724042383596128:
```
Digging in a little deeper, with the same model and parameters as above

```
# this succeeds:
OrbitalElements._α_inner_extremum(n1,n2,model,rc,rmin,rmax)

# this fails with ERROR: DomainError with -0.0002724042383596128:
OrbitalElements._α_inner_extremum(n1,n2,model,rc,0.,Inf)
# same value as above, aha!
```

I read the following lines as always resulting in the MWE, since the line `locrmin, locrmax = min(rmin, locrmin), max(rmax, locrmax)` will always result in `0,Inf`:
https://github.com/JuliaStellarDynamics/OrbitalElements.jl/blob/562de7259a393827c551b6fe8ebe11d63a2c9ae8/src/mappings/resonant/boundaries.jl#L186-L195

In theory, we should somehow protect the user, even if they pick an `rmin,rmax` that will fail. But what's happening here seems to be a simpler issue. Not sure what the best solution is.

**Comment by MathieuRoule** [comment]
Thanks Mike ! Fixed in my last commit:
https://github.com/JuliaStellarDynamics/OrbitalElements.jl/blob/d7f3245c2f1be7f4052083ee9809853230575886/src/mappings/utils/extremise.jl#L118-L120

When refining the search, `_extremise_noedges` was looking outside its given boundaries.
